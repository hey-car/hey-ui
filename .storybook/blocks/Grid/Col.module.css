.col {
  flex: 0 1 auto;
  margin: 0 var(--gap-s) 0 0;
}

.col:last-child {
  margin-right: 0;
}

.span-1 {
  width: calc(100% / 12);
}

.span-2 {
  width: calc(100% / 12 * 2);
}

.span-3 {
  width: calc(100% / 12 * 3);
}

.span-4 {
  width: calc(100% / 12 * 4);
}

.span-5 {
  width: calc(100% / 12 * 5);
}

.span-6 {
  width: calc(100% / 2);
}
