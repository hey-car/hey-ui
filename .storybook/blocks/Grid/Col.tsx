import React, {
  InputHTMLAttributes,
  useCallback,
  ChangeEvent,
  ReactNode,
} from 'react';

import styles from './Col.module.css';

type ColProps = {
  children?: ReactNode;
  className?: string;
  span?: number;
};

export const Col: React.FC<ColProps> = ({
  children,
  className,
  span = 12,
  ...restProps
}) => (
  <div
    className={`${styles.col} ${styles[`span-${span}`]} ${className || ''}`}
    {...restProps}
  >
    {children}
  </div>
);
