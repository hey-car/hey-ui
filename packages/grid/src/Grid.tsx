/* eslint-disable @typescript-eslint/naming-convention */
import Col from './components/Col';
import Container from './components/Container';
import Row from './components/Row';

const Grid = {
  Row,
  Col,
  Container,
  displayName: 'Grid',
} as const;

export default Grid;
