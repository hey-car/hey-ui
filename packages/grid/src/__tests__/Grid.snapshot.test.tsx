import React from 'react';
import { render } from '@testing-library/react';

import Grid from '../Grid';

describe('Grid Snapshots tests', () => {
  it('should match snapshot', () => {
    expect(
      render(
        <Grid.Row>
          <Grid.Col
            width={{
              mobile: 12,
            }}
          >
            Hello
          </Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });

  it('should render row with centered elements', () => {
    expect(
      render(
        <Grid.Row justify="center">
          <Grid.Col width={{ desktop: { m: 12 } }}>Hello</Grid.Col>
          <Grid.Col width={{ desktop: { m: 6 } }}>World!</Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });

  it('should render row with centered elements', () => {
    expect(
      render(
        <Grid.Row justify="center">
          <Grid.Col width={{ mobile: 12 }}>Hello</Grid.Col>
          <Grid.Col width={{ mobile: 3 }}>World!</Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });

  it('should render row with vertically centered elements', () => {
    expect(
      render(
        <Grid.Row align="middle">
          <Grid.Col width={{ mobile: 12 }}>Hello</Grid.Col>
          <Grid.Col width={{ mobile: 3 }}>
            World!<div>Extra text</div>
          </Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });

  it('should render row with row gap', () => {
    expect(
      render(
        <Grid.Row gutter={{ mobile: 8 }}>
          <Grid.Col width={{ mobile: 12 }}>Hello</Grid.Col>
          <Grid.Col width={{ mobile: 3 }}>World!</Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });

  it('should render row with Grid.Col gap', () => {
    expect(
      render(
        <Grid.Row gutter={{ mobile: 12 }}>
          <Grid.Col width={{ mobile: 12 }}>Hello</Grid.Col>
          <Grid.Col width={{ mobile: 3 }}>World!</Grid.Col>
        </Grid.Row>,
      ),
    ).toMatchSnapshot();
  });
});
