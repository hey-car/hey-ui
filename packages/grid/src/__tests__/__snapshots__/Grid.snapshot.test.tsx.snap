// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component between gutter-mobile-s-16 gutter-desktop-m-24"
      >
        <div
          class="col component width-mobile-s-12 width-mobile-m-6"
        >
          Hello
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component between gutter-mobile-s-16 gutter-desktop-m-24"
    >
      <div
        class="col component width-mobile-s-12 width-mobile-m-6"
      >
        Hello
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid Snapshots tests should render row with Grid.Col gap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component between gutter-mobile-12"
      >
        <div
          class="col component width-mobile-12"
        >
          Hello
        </div>
        <div
          class="col component width-mobile-3"
        >
          World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component between gutter-mobile-12"
    >
      <div
        class="col component width-mobile-12"
      >
        Hello
      </div>
      <div
        class="col component width-mobile-3"
      >
        World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid Snapshots tests should render row with centered elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component center gutter-mobile-s-16 gutter-desktop-m-24"
      >
        <div
          class="col component width-desktop-m-12"
        >
          Hello
        </div>
        <div
          class="col component width-desktop-m-6"
        >
          World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component center gutter-mobile-s-16 gutter-desktop-m-24"
    >
      <div
        class="col component width-desktop-m-12"
      >
        Hello
      </div>
      <div
        class="col component width-desktop-m-6"
      >
        World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid Snapshots tests should render row with centered elements 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component center gutter-mobile-s-16 gutter-desktop-m-24"
      >
        <div
          class="col component width-mobile-12"
        >
          Hello
        </div>
        <div
          class="col component width-mobile-3"
        >
          World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component center gutter-mobile-s-16 gutter-desktop-m-24"
    >
      <div
        class="col component width-mobile-12"
      >
        Hello
      </div>
      <div
        class="col component width-mobile-3"
      >
        World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid Snapshots tests should render row with row gap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component between gutter-mobile-8"
      >
        <div
          class="col component width-mobile-12"
        >
          Hello
        </div>
        <div
          class="col component width-mobile-3"
        >
          World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component between gutter-mobile-8"
    >
      <div
        class="col component width-mobile-12"
      >
        Hello
      </div>
      <div
        class="col component width-mobile-3"
      >
        World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid Snapshots tests should render row with vertically centered elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row component middle between gutter-mobile-s-16 gutter-desktop-m-24"
      >
        <div
          class="col component width-mobile-12"
        >
          Hello
        </div>
        <div
          class="col component width-mobile-3"
        >
          World!
          <div>
            Extra text
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row component middle between gutter-mobile-s-16 gutter-desktop-m-24"
    >
      <div
        class="col component width-mobile-12"
      >
        Hello
      </div>
      <div
        class="col component width-mobile-3"
      >
        World!
        <div>
          Extra text
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
